devtools/projects/shell-browser/src/app/content-script.ts

```diff
@@ -29,8 +34,47 @@ export const main = () => {
 
   port.onDisconnect.addListener(handleDisconnect);
 
-  const localMessageBus =
-      new SamePageMessageBus('angular-devtools-content-script', 'angular-devtools-backend');
+  const detectAngularMessageBus = new SamePageMessageBus(
+      `angular-devtools-content-script-${location.href}`,
+      `angular-devtools-detect-angular-${location.href}`);
+
+  detectAngularMessageBus.on('detectAngular', (detectionResult) => {
+    // only install backend once
+    if (backendInstalled) {
+      return;
+    }
+
+    if (detectionResult.isAngularDevTools !== true) {
+      return;
+    }
+
+    if (detectionResult.isAngular !== true) {
+      return;
+    }
+
+    if (detectionResult.isDebugMode !== true) {
+      return;
+    }
+
+    if (detectionResult.isSupportedAngularVersion !== true) {
+      return;
+    }
+
+    // Defensive check against non html page. Realistically this should never happen.
+    if (document.contentType !== 'text/html') {
+      return;
+    }
```

Comment:
```suggestion
if (
  backendInstalled ||
  detectionResult.isAngularDevTools !== true ||
  detectionResult.isAngular !== true ||
  detectionResult.isDebugMode !== true ||
  detectionResult.isSupportedAngularVersion !== true ||
  // Defensive check against non html page. Realistically this should never happen.
  document.contentType !== 'text/html'
) {
  return;
}
```

And in case the formatter  doesn't keep 1 condition per line, you can add a trailing comment to force this layout !

Reply:
Personally I prefer multiple `if` statements so I don't have to try to under the `||` combinations and each can get its own comments as necessary. I don't feel that strongly about it though.

Reply:
I prefer the multiple as well. 1 if statement with multiple unrelated criteria can get difficult to parse visually as combinations of `&&`, `||`, `===`, `!==` start getting used.